lib/synapse.rb

```diff
@@ -78,7 +78,7 @@ def run
             raise "synapse: service watcher #{w.name} failed ping!" unless alive
           end
 
-          if @config_updated.get
+          if @config_updated.get_and_set(false)
```

Comment:
I feel a little bit confused by doing this. I prefer simply `get` first and then `set` later.

Reply:
Discussed offline; this is necessary to prevent a race condition when multiple `notifications` are sent in quick succession